app/models/order.rb

```diff
@@ -485,7 +511,7 @@ def email_properties
   private
 
   def assign_code
-    self.code = Order.generate_gc_code if goodcity_order?
+    self.code = Order.generate_gc_code(detail_type) if goodcity_order?
```

Comment:
You can rename that method into `generate_code`, since it no longer deals exclusively with gc orders